content/blog/open-sourcing-workshops-on-demand-part-5-create-a-workshop.md

```diff
@@ -0,0 +1,236 @@
+
+A﻿s an admin of the Workshops-on-Demand infrastructure, I had to perform several tasks:
+
+##### O﻿n the backend server:
+
+The  workshop will require:
+
+* A dedicated server running Linux to allow the student to run Ansible playbooks against during the workshops' labs.
+
+  * This means preparing a server (VM or physical) : We will consider it as an appliance
+  * Updating the relevant variable file to associate the IP address of the server to the workshop (there could be multiple servers too associated to a given workshop)
+* A﻿ set of scripts under wod-backend/scripts or wod-private/scripts folders depending on the nature of the workshop to manage the workshop's lifecycle
+
+  * Some generic scripts applying to all workshops' appliances : general setup phase setting up common requirements for any appliance (student users creation, ssh keys, etc..) up to some specific ones dedicated to a given workshop
+
+    * create-appliance.sh (ssh keys, ldap setup)
+    * setup-appliance.sh \[WKSHP-NAME] (Student setup, Appliance Setup, Workshop setup) calls:
+
+      * setup-globalappliance.sh (global / generic setup)
+      * setup-\[WKSHP-NAME].sh (Prepare appliance with workshop's reqs, Docker image for instance)
+    * create-\[WKSHP-NAME].sh (called at deployement time to instantiate the necessary appliance(s) requiered by the workshop 
+    * reset-appliance (reset ssh keys and students credentials on appliance
+    * cleanup-\[WKSHP-NAME].sh (takes care of cleanup some workshop's specifics)
+    * reset-\[WKSHP-NAME].sh (reset of the workshop's appliance, docker compose down of a container for instance)
+* A set of variables to be leveraged by the notebooks. These variables are to be set in yml format. They will be parsed at deployment time to set student ids, appliance IP addresses, and other relevant parameters like ports, or simulated hardware information
+
+Whenever all the scripts are functional and that the necessary actions have been performed both on backend and frontend servers, some functional tests can be conducted using cli and later webui as described earlier for the simple workshop example.
+
+Testing the workshop: 
+
+* One can leverage the wod-test-action.sh script to test a workshop lifecycle action from deployment (CREATE) to CLEANUP, RESET, or PURGE.
+
+  ```shell
+  dev@dev3:~$ wod-test-action.sh
+  Syntax: wod-test-action.sh <CREATE|CLEANUP|RESET|PURGE|PDF|WORD> WKSHOP [MIN[,MAX]
+  ACTION is mandatory
+  ```
+
+`Note: The available trace under ~/.mail/from will detail the different steps of the action and allow you to troubelshoot any issue.`
+
+When all tests have validated the workshop, it can follow the move to prod cycle.
+
+You should now have a better understanding of the necessary tasks associated to the creation of a workshop. As you can see, it requires steps on the various sides of the infrastructure.
```

Comment:
Change the last part of the second sentence to:

As you can see, there are a number of steps required on the various sides of the infrastructure (architecture might be a better word).